plugin/src/it/advanced/frontend/src/test/java/com/spotify/it/frontend/MainIT.java

```diff
@@ -20,90 +20,111 @@
-import static org.hamcrest.Matchers.containsString;
-import static org.hamcrest.Matchers.describedAs;
-import static org.hamcrest.Matchers.is;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
 
 public class MainIT {
+  private static final Logger log = LoggerFactory.getLogger(MainIT.class);
+
+  private static final int BACKEND_PORT = 1337;
+  private static final int FRONTEND_PORT = 1338;
+
+  /**
+   * Create a Network that both containers are attached to. When the containers are setup, they have
+   * .withNetworkAliases("foo") set, which allows the other container to refer to http://foo/ when
+   * one container needs to talk to another.
+   *
+   * This is needed because container.getContainerIpAddress() is only for use for a test
+   * to communicate with a container, not container-to-container communication -
+   * getContainerIpAddress() will return "localhost" typically
+   */
+  @Rule public final Network network = Network.newNetwork();
 
-  @ClassRule
-  public static final HeliosDeploymentResource HELIOS = new HeliosDeploymentResource(
-      HeliosSoloDeployment.fromEnv()
-          .checkForNewImages(true)
-          .removeHeliosSoloOnExit(true)
-          .build());
+  @Rule
+  public final GenericContainer backendJob = createBackend(network);
 
   @Rule
-  public final TemporaryJobs temporaryJobs = TemporaryJobs.builder()
-      .client(HELIOS.client())
-      .deployTimeoutMillis(TimeUnit.MINUTES.toMillis(1))
-      .build();
+  public final GenericContainer frontendJob = createFrontend(network);
+
+  private GenericContainer createBackend(final Network network) {
+    final String image;
+    try {
+      image = Resources.toString(
+          Resources.getResource("META-INF/docker/com.spotify.it/backend/image-name"),
+          Charsets.UTF_8).trim();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    final GenericContainer container = new GenericContainer(image)
+        .withExposedPorts(BACKEND_PORT)
+        .withNetwork(network)
+        .withNetworkAliases("backend")
+        .waitingFor(Wait.forHttp("/api/version"));
+
+    // start early, since frontend needs to know the port of backend
+    container.start();
+
+    return container;
+  }
+
+  private GenericContainer createFrontend(final Network network) {
+    final String image;
+    try {
+      image = Files.readFirstLine(new File("target/docker/image-name"), Charsets.UTF_8);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    return new GenericContainer(image)
+        .withExposedPorts(1338)
+        .withCommand("http://backend:" + BACKEND_PORT)
```

Comment:
What is the point of this command? It is just a uri.

Reply:
the frontend container expects it as an argument to know how to talk to the backend container: https://github.com/spotify/dockerfile-maven/blob/2738a95be732af76fde5be8f0b43fcba2235acbf/plugin/src/it/advanced/frontend/src/main/java/com/spotify/it/frontend/Main.java#L37-L42